web/src/templates.rs

```diff
@@ -0,0 +1,86 @@
+use std::collections::HashMap;
+
+use axum::{
+    extract::{FromRequestParts, Query},
+    response::{Html, IntoResponse, Response},
+};
+use maud::PreEscaped;
+
+use crate::{err, AppState};
+
+pub struct Flash {
+    error: Option<String>,
+}
+
+#[async_trait::async_trait]
+impl FromRequestParts<AppState> for Flash {
+    type Rejection = err::Error;
+
+    async fn from_request_parts(
+        parts: &mut axum::http::request::Parts,
+        _state: &AppState,
+    ) -> Result<Self, Self::Rejection> {
+        let Query(query) =
+            Query::<HashMap<String, String>>::from_request_parts(parts, _state).await?;
+        let error = query.get("flash[error]").cloned();
+
+        Ok(Flash { error })
+    }
+}
+
+pub struct Template {
+    flash: Flash,
```

Comment:
Inside our template we have these little `flashes`. These are why we made the Template struct over the method we had when we paired

I wanted to be able to send a simple message from one page to the next to be displayed to the user as an error

So the tempalte now looks for a `?flash` query param that it can pull out and put in this `Flash` struct

When we render the template if there is an error flash we render it before the main page content! We use this for instance when you try to signup with an existing username

Reply:
Does that have to do anything with our flash crate?

Reply:
We don't have a crate for `flash` but it does related to the `Flash` struct I added!

The `Flash` struct is used to old the query params I mention here once they are de-serialized!